Add "Image Inserter" community plugin

cloudy9101 opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/cloudy9101/obsidian-image-inserter

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
You can remove `obsidian` from the plugin id, its unecessary.

[id: 'image-inserter-plugin',](https://github.com/cloudy9101/obsidian-image-inserter/blob/10f74b7aac144e4a46b98d121a4e8ecbb7f3aea1/main.ts#L24)
I would use a different id, like `insert-unsplash-image`, you don't need to add the plugin name to the command id, that is already handled by Obsidian.

[const url = new URL("/search/photos", "https://insert-unsplash-image.cloudy9101.com/")](https://github.com/cloudy9101/obsidian-image-inserter/blob/10f74b7aac144e4a46b98d121a4e8ecbb7f3aea1/fetcher.ts#L15)
I guess you are using your own server, so users don't have to have an API key for unsplash?
Please document this in the README.
I would also recommend adding some code to handle errors from your own API, since at the moment it would just fail without any notice.

[await fetch(url)](https://github.com/cloudy9101/obsidian-image-inserter/blob/10f74b7aac144e4a46b98d121a4e8ecbb7f3aea1/fetcher.ts#L21)
Use [`requestUrl`](https://github.com/obsidianmd/obsidian-api/blob/1b4f6e2e5753c8ac2d538bc5d55c38f228450cb2/obsidian.d.ts#L2894-L2898) instead.

cloudy9101 commented:
Thank you for the review. I updated the code.

1. Plugin id changed to `insert-unsplash-image`
2. Document the proxy in the plugin repo's README
3. Notice there's something went wrong when the API request failed
4. Replaced `fetch` to `requestUrl`

Please review again when you got time, thanks.
